### ISSUE TITLE
Cannot read property 'defaultPass' of undefined

### ISSUE THREAD

**Issue by crhym3** [GoogleChrome/lighthouse#527]
```
...
Tue, 26 Jul 2016 21:05:33 GMT statusEnd Gathering: trace
Tue, 26 Jul 2016 21:05:34 GMT method <= browser OK Runtime.evaluate 
Tue, 26 Jul 2016 21:05:34 GMT statusEnd Gathering: Accessibility
Tue, 26 Jul 2016 21:05:34 GMT status Gathering: ScreenshotFilmstrip
Tue, 26 Jul 2016 21:05:34 GMT statusEnd Gathering: ScreenshotFilmstrip
Tue, 26 Jul 2016 21:05:34 GMT status Gathering: CriticalRequestChains
Tue, 26 Jul 2016 21:05:34 GMT statusEnd Gathering: CriticalRequestChains
Tue, 26 Jul 2016 21:05:34 GMT status Gathering: Speedline
Runtime error encountered: TypeError: Cannot read property 'defaultPass' of undefined
    at Speedline.afterPass (/home/moonlight/node_modules/lighthouse/lighthouse-core/driver/gatherers/speedline.js:26:40)
    at chain.then._ (/home/moonlight/node_modules/lighthouse/lighthouse-core/driver/index.js:139:45)
TypeError: Cannot read property 'defaultPass' of undefined
    at Speedline.afterPass (/home/moonlight/node_modules/lighthouse/lighthouse-core/driver/gatherers/speedline.js:26:40)
    at chain.then._ (/home/moonlight/node_modules/lighthouse/lighthouse-core/driver/index.js:139:45)
```

lighthouse was installed with `npm install googlechrome/lighthouse#d07719e7`.
node v6, npm v3

**Comment by crhym3** [comment]
@surma could it be that it has something to do with your commit at https://github.com/GoogleChrome/lighthouse/commit/eaede963c7c7a73ca6facb6537865eb599d0572a?

**Comment by surma** [comment]
Yes definitely related to that. Will take a stab at that in the morning.

**Comment by paulirish** [comment]
I've reverted that commit in the meantime. We'll have to land it in another go, @surma 
master should be good for you now, @crhym3 

---

@surma I'm pretty sure this is because the speedline needs the trace in `afterPass` but the `traces[...]` assignment happens directly *after* that: https://github.com/GoogleChrome/lighthouse/commit/eaede963c7c7a73ca6facb6537865eb599d0572a#diff-1decc169d5e091afb56f6dbd0dd2564fR191

**Comment by brendankenny** [comment]
we should make sure to check why the tests didn't catch this. Overly permissive on mock data, maybe?

**Comment by surma** [comment]
Fixed w/ #531
